sematic/db/queries.py

```diff
@@ -200,6 +200,22 @@ def get_resources_by_root_id(root_run_id: str) -> List[ExternalResource]:
         return list(set(r[0] for r in results))
 
 
+def get_external_resources_by_run_id(run_id: str) -> List[ExternalResource]:
+    """
+    Get the external resources used by a run
+    """
+    with db().get_session() as session:
+        result = (
+            session.query(RunExternalResource, ExternalResource)
+            .filter(RunExternalResource.run_id == run_id)
+            .join(
+                ExternalResource, ExternalResource.id == RunExternalResource.resource_id
+            )
+            .all()
+        )
+    return [row[ExternalResource] for row in result]
```

Comment:
I believe this simpler, more efficient query should work:
```suggestion
            session.query(ExternalResource)
            .join(
                ExternalResource, ExternalResource.id == RunExternalResource.resource_id
            )
            .all()
        )
    return result
```

Reply:
Hey @tscurtu 

I guess what you meant is probably 
```
session.query(ExternalResource)
            .join(RunExternalResource, run_id == RunExternalResource.run_id)
            .join(
                ExternalResource, ExternalResource.id == RunExternalResource.resource_id
            )
            .all()
```
otherwise there is no usage of input parameter `run_id`, then we are not filtering anything.

But unfortunately that doesn't work either. SQLAlchemy complains that the same table `ExternalResource` is joined twice. 

But your approach gives me great inspiration. I found the below works, and it's probably what you intended:

```
session.query(ExternalResource)
            .join(
                RunExternalResource, sqlalchemy.and_(
                    run_id == RunExternalResource.run_id,
                    ExternalResource.id == RunExternalResource.resource_id
                )
            )
            .all()
```